### PR TITLE
D3ASIM-1645: Populating cumulative offers/bids for PAC line_sweep alg…

### PR DESCRIPTION
**Pull request by faizan2590**
…orithm and making plots compatable when past_markets are deleted.

### CODE REVIEW

File: src/d3a/d3a_core/export.py [hunk 1]
```diff
@@ -553,12 +556,12 @@ def _plot_supply_demand_curve(self, subdir: str, area: Area):
                                           name=time_slot.format(DATE_TIME_FORMAT +
                                                                 ' Clearing-Energy'))
                     data.append(data_obj)
-                    xmax = max(xmax, clearing_point[1]) * 3
+                    xmax = max(xmax, clearing_point[1]) * 6
```
Comment (spyrostz) [inline src/d3a/d3a_core/export.py]: I think I already asked you offline, but can you remind me why do we need the hardcoded value here ?

Comment (faizan2590) [reply src/d3a/d3a_core/export.py]: Ohkay, can't recall, should I move it to ConsSettings?

File: src/d3a/d3a_core/sim_results/file_export_endpoints.py [hunk 2]
```diff
@@ -106,6 +109,21 @@ def _get_stats_from_market_data(self, out_dict, area, balancing):
     def update_plot_stats(self, area):
         self._get_stats_from_market_data(self.plot_stats, area, False)
         self._get_stats_from_market_data(self.plot_balancing_stats, area, True)
+        if ConstSettings.IAASettings.MARKET_TYPE == 3:
+            if len(area.past_markets) == 0:
```
Comment (hannesdiedrich) [inline src/d3a/d3a_core/sim_results/file_export_endpoints.py]: I would vote for refactoring the following lines into a separate function.

File: src/d3a/d3a_core/export.py (hunk 1 above)
Comment (spyrostz) [reply src/d3a/d3a_core/export.py]: Leave it as it is, not that big of a deal, I am just curious why did you need to change 3 -> 6

Comment (faizan2590) [reply src/d3a/d3a_core/export.py]: I was playing around with supply/demand curve, with `3` wasn't able to fit the whole supply/demand curve wrt my setup_file `test_clearing_energy`

File: src/d3a/setup/two_sided_pay_as_clear/test_clearing_energy.py [hunk 3]
```diff
@@ -41,22 +39,32 @@ def get_setup(config):
             Area(
                 'House 1',
                 [
-                    Area('H1 General Load', strategy=LoadHoursStrategy(avg_power_W=100,
-                                                                       hrs_per_day=24,
-                                                                       hrs_of_day=list(
-                                                                           range(24)),
-                                                                       final_buying_rate=35),
+                    Area('H1 General Load1', strategy=LoadHoursStrategy(
```
Comment (hannesdiedrich) [inline src/d3a/setup/two_sided_pay_as_clear/test_clearing_energy.py]: Sorry, I dont understand why these changes to the setup file are needed. Can you please explain? Maybe also offline!?

File: src/d3a/d3a_core/export.py (hunk 1 above)
Comment (spyrostz) [reply src/d3a/d3a_core/export.py]: Got it, thanks!
